test/test_query.cpp

```diff
@@ -5773,4 +5772,339 @@ TEST_TYPES(Query_IntCompressed, Equal, NotEqual, Less, LessEqual, Greater, Great
     }
 }
 
+// Many of our tests just test the correctness of sorting strings.
+// For compressed strings we can use the string ids to perform the
+// same task. We just need to commit first and then run the query.
+// These tests are mainly verifying the following:
+//
+//  1. Store N strings inside a Mixed. Commit and sort.
+//  2. Store inside a Mixed integers and Strings. Commit and sort
+//  3. Store N strings in compressed format inside a Mixed property, store another N strings uncompressed in another
+//  column. Sort using both columns.
+//  4. Store N compressed strings inside a table, linked by another table. Sort over links.
+
+// Note: Strings and Mixed use the same logic for compressing strings. Thus these tests are solely using Mixed
+// columns.
+
+
+static int gen_random_int(int min = 1, int max = 100)
+{
+    std::random_device rd;
+    std::mt19937 generator(rd());
+    std::uniform_int_distribution<> distribution(min, max);
+    return distribution(generator);
+}
+
+static std::string gen_random_string(size_t length)
```

Comment:
Good idea to test random strings. I'd love to see some utf8 strings tested as well. My suggestion would be to look at the tests such as `UTF8_Compare_Strings` (and the others nearby) where you could modify the `str_compare` lambda to use compression as well and verify that both modes produce the same sort order.